types/valuerenderer/valuerenderer.go

```diff
@@ -0,0 +1,206 @@
+package valuerenderer
+
+import (
+	"context"
+	"errors"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/dustin/go-humanize"
+
+	"github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// ValueRenderer defines an interface to produce formated output for Int,Dec,Coin types as well as parse a string to Coin or Uint.
+type ValueRenderer interface {
+	Format(context.Context, interface{}) (string, error)
+	Parse(context.Context, string) (interface{}, error)
+}
+
+// denomQuerierFunc takes a context and a denom as arguments and returns metadata, error
```

Comment:
let's update a comment to something more meaningful. We know the types, instead, describe what the function should do.

Reply:
I hope it sounds better: `denomQuerierFunc` should take `context` and `baseDenom` to invoke `DenomMetaData` method on `queryClient`. This method will return`banktypes.Metadata`.